include/aspect/utilities.h

```diff
@@ -813,6 +813,42 @@ namespace aspect
          */
         std_cxx11::unique_ptr<aspect::Utilities::AsciiDataLookup<1> > lookup;
     };
+
+    /**
+     * A weighted p norm average caluclator:
+     * $\text{weighed\_p\_norm\_average} =
+     * \frac{1}{k}(\sum_{i=1}^k \big(\text{weights}_i \text{values}_i^p)\big)^{\frac{1}{p}}$.
+     *
+     * There have been several special cases implemented, mostly for performance reasons, and
```

Comment:
Oh,  I see what you mean.

That is an implementation detail. For the user, *how the function is implemented* doesn't actually matter, and it shouldn't be in the documentation. What matters for the user reading the documentation is *what the function does*.